okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -1049,6 +1071,29 @@ open class OkHttpClient internal constructor(
       this.minWebSocketMessageToCompress = bytes
     }
 
+    /**
+     * Set an override for host to allow insecure connections, typically a devserver with
+     * a self-signed certificate unavailable to the dev build of the client.
+     *
+     * This will override any existing sslSocketFactory with the platform default.
+     *
+     * n.b. not for production use, only available from Android version 26.
```

Comment:
We can make this work on all versions, can’t we?

Reply:
We can implement it working, but Android only calls the method with an additional host on some versions.  I suspect it's actually version 23 -> 24.

e.g. tied in with

```

  override fun isCleartextTrafficPermitted(hostname: String): Boolean {
    return try {
      val networkPolicyClass = Class.forName("android.security.NetworkSecurityPolicy")
      val getInstanceMethod = networkPolicyClass.getMethod("getInstance")
      val networkSecurityPolicy = getInstanceMethod.invoke(null)
      api24IsCleartextTrafficPermitted(hostname, networkPolicyClass, networkSecurityPolicy)
    } catch (_: ClassNotFoundException) {
      super.isCleartextTrafficPermitted(hostname)
    } catch (_: NoSuchMethodException) {
      super.isCleartextTrafficPermitted(hostname)
    } catch (e: IllegalAccessException) {
      throw AssertionError("unable to determine cleartext support", e)
    } catch (e: IllegalArgumentException) {
      throw AssertionError("unable to determine cleartext support", e)
    } catch (e: InvocationTargetException) {
      throw AssertionError("unable to determine cleartext support", e)
    }
  }
```

Reply:
https://github.com/aosp-mirror/platform_frameworks_base/commit/93962c2f0118bcda78ffde7e0b6be73da7ab2500#diff-9c086e3a85d0f86ba44bf971ebca1171

https://github.com/aosp-mirror/platform_frameworks_base/commit/0f5d386a769cafa3f4faf93f17acb22503b6fd3e#diff-9c086e3a85d0f86ba44bf971ebca1171

Reply:
See https://github.com/square/okhttp/pull/5960